mysql/data.go

```diff
@@ -0,0 +1,250 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mysql
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/bits"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/civil"
+	"cloud.google.com/go/spanner"
+
+	"github.com/cloudspannerecosystem/harbourbridge/internal"
+	"github.com/cloudspannerecosystem/harbourbridge/schema"
+	"github.com/cloudspannerecosystem/harbourbridge/spanner/ddl"
+)
+
```

Comment:
Yes sure. Main differences are just the way empty columns are represented ("\\N" - postgres , "<nil>" - mysql), convTimestamp and convArray.
For refactoring, we can change convTimeStamp (we can think of some helper function to either convert offset to Timezone ID - mysql or convert Timezone ID to offset - postgres), add 2 values for representing empty columns and convArray is good because MySQL is only using the string case. Then we can club everything in just 1 data.go file in `internal` package. Basically, we can think a way to convert functions such that specifying driver is not required, still function would be able to handle both the cases. Also, using go interface would be better for future enhancements. 

We can discuss more about this, for now I will put this comment on top of the file.